src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs

```diff
@@ -71,6 +71,38 @@ protected static bool IsNonIntersectingNamespace(ISymbol recommendationSymbol, S
                                               declarationSyntax.Span.IntersectsWith(candidateLocation.SourceSpan)));
         }
 
+        /// <summary>
+        /// If container is a tuple type, any of its tuple element which has a friendly name will cause
+        /// the suppression of the corresponding default name (ItemN).
+        /// In that case, Rest is also removed.
+        /// </summary>
+        protected static IEnumerable<ISymbol> SuppressDefaultTupleElements(INamespaceOrTypeSymbol container,
+            IEnumerable<ISymbol> symbols)
+        {
+            if (!container.IsType)
+            {
+                return symbols;
+            }
+
+            var type = (ITypeSymbol)container;
+            if (!type.IsTupleType)
+            {
+                return symbols;
+            }
+
+            var tuple = (INamedTypeSymbol)type;
```

Comment:
To remove Rest, you would need to filter by name, unfortunately. 
The "right" way of doing that, IMO, would be making TupleElementIndex API public and just filtering out everything with -1 index.